[pkg] add a debug pickler

PaliC opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #70967
* #70966
* #70965
* __->__ #70646
* #70642
* #70641

Adds a stack trace to the pickler in order to help locate deeply nested unpicklable objects. This pr specifically sets up testing and the overarching stack trace (as well as the objects that will be used in it). The debug pickler is also able to find unpicklable objects in dicts.

Differential Revision: [D33414363](https://our.internmc.facebook.com/intern/diff/D33414363)

PaliC committed:
[pkg] add a debug pickler

[ghstack-poisoned]

PaliC committed:
Update on "[pkg] add a debug pickler"




[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33414363).

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33414363).

PaliC committed:
Update on "[pkg] add a debug pickler"


Adds a stack trace to the pickler in order to help locate deeply nested unpicklable objects.

Differential Revision: [D33414363](https://our.internmc.facebook.com/intern/diff/D33414363)

[ghstack-poisoned]

PaliC committed:
Update on "[pkg] add a debug pickler"


Adds a stack trace to the pickler in order to help locate deeply nested unpicklable objects.

Differential Revision: [D33414363](https://our.internmc.facebook.com/intern/diff/D33414363)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33414363).

PaliC committed:
Update on "[pkg] add a debug pickler"


Adds a stack trace to the pickler in order to help locate deeply nested unpicklable objects. This pr specifically sets up testing and the overarching stack trace (as well as the objects that will be used in it). The debug pickler is also able to find unpicklable objects in dicts.

Differential Revision: [D33414363](https://our.internmc.facebook.com/intern/diff/D33414363)

[ghstack-poisoned]

PaliC committed:
Update on "[pkg] add a debug pickler"


Adds a stack trace to the pickler in order to help locate deeply nested unpicklable objects. This pr specifically sets up testing and the overarching stack trace (as well as the objects that will be used in it). The debug pickler is also able to find unpicklable objects in dicts.

Differential Revision: [D33414363](https://our.internmc.facebook.com/intern/diff/D33414363)

[ghstack-poisoned]

PaliC committed:
Update on "[pkg] add a debug pickler"


Adds a stack trace to the pickler in order to help locate deeply nested unpicklable objects. This pr specifically sets up testing and the overarching stack trace (as well as the objects that will be used in it). The debug pickler is also able to find unpicklable objects in dicts.

Differential Revision: [D33414363](https://our.internmc.facebook.com/intern/diff/D33414363)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33414363).

suo commented on torch/package/_package_pickler.py:
```diff
@@ -93,6 +117,178 @@ def save_global(self, obj, name=None):
     dispatch[FunctionType] = save_global
 
 
+class DebugInfo:
```
Let's split this out into a separate file

suo reviewed (approved):
nice!

suo reviewed (approved):
nice :)

PaliC committed:
Update on "[pkg] add a debug pickler"


Adds a stack trace to the pickler in order to help locate deeply nested unpicklable objects. This pr specifically sets up testing and the overarching stack trace (as well as the objects that will be used in it). The debug pickler is also able to find unpicklable objects in dicts.

Differential Revision: [D33414363](https://our.internmc.facebook.com/intern/diff/D33414363)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33414363).
